Apply comments and merge both branches up to date

anpineda opened the pull request:
Apply comments and merge both branches up to date

luvalverde commented on README.md:
```diff
@@ -1,6 +0,0 @@
-# local_match
```
Why are you deleting this file?

luvalverde commented on LICENSE:
```diff
@@ -1,674 +0,0 @@
-
-  Each time you convey a covered work, the recipient automatically
-receives a license from the original licensors, to run, modify and
-propagate that work, subject to this License.  You are not responsible
```
Why are you deleting this file?

luvalverde commented on git-pre-commit-format:
```diff
@@ -1,24 +0,0 @@
-#!/bin/bash
```
Why are you deleting this file?

luvalverde commented on localMatchApp/headers/classevent.h:
```diff
@@ -0,0 +1,83 @@
+
+    /*! Declaration of the variable that saves the id of the event */
+    int idEvent;
+    /*! Declaration of the variable that saves the total of the events */
```
Avoid writting code all wrapped together.
Use blank lines between each variable.
Apply this comment in all your PR.

luvalverde commented on localMatchApp/headers/classevent.h:
```diff
@@ -0,0 +1,83 @@
+ * be able to develop the functions
+ * to manage the variables of the event
+ */
+class ClassEvent
```
According with the coding standard, files defining a class must be named as the class (ClassEvent.h).
Apply this comment in all your PR.

luvalverde commented on localMatchApp/headers/classevent.h:
```diff
@@ -0,0 +1,83 @@
+    QString pathImageEvent;
+
+    /*!
+     * The functions to set and get the id of the event
```
Each function must has its own description.
Apply this comment in all your PR.

luvalverde commented on localMatchApp/main.cpp:
```diff
@@ -1,10 +1,11 @@
-  w.show ( );
+int main(int argc, char *argv[])
+{
+    QApplication a(argc, argv);
```
Improve your variable names (a, w)

luvalverde commented on localMatchApp/sources/classevent.cpp:
```diff
@@ -0,0 +1,88 @@
+}
+
+/*!
+ * The functions to set the id of the event
```
.cpp files should not contain functions descriptions. It is only at .h files.
Apply this comment in all your PR.

luvalverde commented on localMatchApp/sources/functionsevent.cpp:
```diff
@@ -0,0 +1,42 @@
+    finalDateEvent = "14/12/18 13:00";
+    pathImageEvent = ":/images/eventExample.svg";
+
+    ClassEvent event;
```
Add a comment describing what this does

luvalverde commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,140 @@
+    ui->setupUi(this);
+
+    /*! The variable index is initialized with zero because the carousel
+     * will begin with the fisrt position */
```
... the first ...

anpineda commented:
I already add the files that were delete

Sorry but not sorry
PD: XOXO

luvalverde commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,140 @@
+  connect (ui->ButtonNext, SIGNAL (clicked ()), this, SLOT (onButtonNextClicked ()));
+  connect (ui->ButtonPrevious, SIGNAL (clicked ()), this, SLOT (onButtonPreviousClicked ()));
+  connect (ui->pushButtonSubscribe, SIGNAL (clicked ()), this, SLOT (onPushButtonSubscribeClicked ()));
+
```
remove extra blank lines

luvalverde commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,140 @@
+    ui->pictureLogo->setPixmap (pixLogo);
+
+    ///@TODO Remove the name user value default and set the data by the DB
+    ui->labelNameUser->setText ("User Name");
```
use a variable for the user name, so we then only need to update this variable value with the DB info

luvalverde commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,140 @@
+        "rgb(187, 205, 225);" );
+    ui->labelDescription->setWordWrap (true);
+
+  nameEvent = event.getNameEvent();
```
Add comments

luvalverde commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,140 @@
+  finalDateEvent=event.getFinalDateEvent();
+  pathImageEvent = event.getPathImageEvent();
+
+  connect (ui->ButtonNext, SIGNAL (clicked ()), this, SLOT (onButtonNextClicked ()));
```
Add comments

luvalverde commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,140 @@
+
+void HomeWindow::onPushButtonSubscribeClicked()
+{
+    ui->listWidget->addItem(new QListWidgetItem(QIcon(pathImageEvent),nameEvent));
```
Add comments

luvalverde commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,140 @@
+    eventStatus = QString::number(i_indexOfCarousel+1)+"/"+QString::number(event.getTotalEvents())+" of events";
+
+    /*! If there are no events a message is show to the user */
+    if(event.getTotalEvents() == NOEVENTS){
```
According with the coding standard, braces must be on separate lines.
Apply this in all your PR

luvalverde commented on localMatchApp/sources/mainwindow.cpp:
```diff
@@ -13,42 +13,50 @@
+
+void MainWindow::onButtonSignUpClicked()
+{
+    SignUp signUpWindow;
```
Add comments

luvalverde commented on localMatchApp/sources/mainwindow.cpp:
```diff
@@ -13,42 +13,50 @@
+
+void MainWindow::onButtonLoginClicked()
+{
+    this->hide();
```
Add comments

luvalverde commented on localMatchApp/sources/signup.cpp:
```diff
@@ -1,44 +1,56 @@
+{
+    const int FIXEDWIDTHSIGNUPWINDOW = 500;
+    const int FIXEDHEIGHTSIGNUPWINDOW = 450;
+    ui->setupUi(this);
```
Avoid writting your code all wrapped together without blank lines.
It is more difficult to read

luvalverde commented on localMatchApp/sources/signup.cpp:
```diff
@@ -1,44 +1,56 @@
+
+void SignUp::onNewUserButtonClicked()
+{
+    QMessageBox messageNewUser;
```
Add comments

luvalverde commented on localMatchApp/headers/classevent.h:
```diff
@@ -0,0 +1,83 @@
+class ClassEvent
+{
+public:
+    ClassEvent();
```
According with the coding standard, all classes must have:
- copy constructor
- copy assignment operator
- destructor

Apply this comment in all your PR

luvalverde reviewed (changes requested):
I added some comments.
Please improve your commit messages.

luvalverde commented:
> I already add the files that were delete
> 
> Sorry but not sorry
> PD: XOXO

But I am sorry for your English grammar @anpineda :)

wfallascr commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,89 @@
+    /*! Declaration of the variable that saves the total of the events */
+    int totalEvents;
+
+    /*! Declaration of the variable that saves the name of the event */
```
Please, provide more meaningful comments. "Declaration of the variable that" is unnecessary.

wfallascr commented on localMatchApp/headers/FunctionsEvent.h:
```diff
@@ -0,0 +1,45 @@
+#ifndef FUNCTIONS_H
+#define FUNCTIONS_H
```
Functions?

wfallascr commented on localMatchApp/headers/FunctionsEvent.h:
```diff
@@ -0,0 +1,45 @@
+ * be able to develop the functions
+ * that will manage the database
+ */
+class FunctionsEvent
```
This name does not reflect the utility of the class. Please, think on a new name for this class.

wfallascr commented on localMatchApp/headers/homewindow.h:
```diff
@@ -2,24 +2,76 @@
+ * The functions of button actions are declared so
+ * they can be used later
+ */
+private slots:
```
Why private slots?

masolano replied:
Because they have to be of the slot type since the signals of the buttons need it to be able to give functionality to the button, if they are only private it does not work

wfallascr commented on localMatchApp/headers/homewindow.h:
```diff
@@ -2,24 +2,76 @@
 
- private:
+private:
   Ui::HomeWindow *ui;
```
Why you use a raw pointer here?

masolano replied:
That part the program creates it alone, it is automatic

wfallascr commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,155 @@
+    indexOfCarousel = 0;
+
+    /*! The carousel function is called and the index value is sent */
+    manageCarousel(indexOfCarousel);
```
What exactly does this function? Manage is a too broad word.

wfallascr commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,155 @@
+     * If the user click the button of subscribe, a new item of local event is
+     * added to the list widget of my events
+     */
+    ui->listWidget->addItem(new QListWidgetItem(QIcon(event.getPathImageEvent()),
```
Where is the _delete_ partner of this _new_?

wfallascr commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,155 @@
+     *  type event to be able to call the get function the values ​​return
+     */
+   FunctionsEvent functionOfEvent;
+   event = functionOfEvent.selectEvent(i_indexOfCarousel);
```
Indentation problem? @Robchaves Please check this

wfallascr commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,155 @@
+    }
+    else{
+        /*! If there are only one event the next and previous buttons are disabled */
+        if (event.getTotalEvents() == ONEEVENT)
```
Ths logic is still too big. Try creating smaller functions.

wfallascr commented:
@anpineda You changed the name of some files (added .txt), please return them to their original name

masolano commented:
PR back to you

luvalverde commented on localMatchApp/CMakeLists.txt:
```diff
@@ -6,6 +6,9 @@ PROJECT(local_match)
 # Enable standard C++ 14 compiler
 SET(CMAKE_CXX_STANDARD 14)
 
+SET (CMAKE_CXX_STANDARD 14)
```
Remove this line.
It already exists in master

wfallascr commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+
+    ~ClassEvent();
+
+    /*! Copy Constructor of class ClassEvent */
```
This comment is unnecessary

wfallascr commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+
+    ClassEvent(const ClassEvent &Event);
+
+    /*! Copy Assigment of class ClassEvent */
```
This comment is unnecessary

wfallascr commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+ * be able to develop the functions
+ * to manage the variables of the event
+ */
+class ClassEvent
```
Name of this class is not good. It is evident that this is a class, so you do not need to add the word "Class". Think of something a little bit more descriptive, SocialEvent, GeneralEvent, UserEvent, etc

wfallascr commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+#include <QWidget>
+
+/*!
+ * The class of Event is created to
```
This comment is out of date

wfallascr commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+#ifndef EVENTS_H
+#define EVENTS_H
```
Guard's name is out of date

wfallascr commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+    QString getNameEvent();
+
+    /*!
+     * The functions to set the description of the event
```
This is clearly a function, you do not need to restate that in the comment. Same for all your comments

wfallascr commented on localMatchApp/sources/ClassEvent.cpp:
```diff
@@ -0,0 +1,101 @@
+
+ClassEvent::~ClassEvent(){}
+
+ClassEvent::ClassEvent(const ClassEvent &Event)
```
The variable name should not start with a capital letter. Event -> event

wfallascr commented on localMatchApp/sources/ClassEvent.cpp:
```diff
@@ -0,0 +1,101 @@
+
+void ClassEvent::copyObjectClassEvent(const ClassEvent &Event)
+{
+    idEvent = Event.idEvent;
```
This is a trivial constructor. [Read](https://en.cppreference.com/w/cpp/language/copy_constructor)

luvalverde commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+
+    /*  Function which initializes an object
+     * using another object of the same class */
+
```
remove extra blank line

luvalverde commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+    void copyObjectClassEvent (const ClassEvent &Event);
+
+    /*!
+     * The functions to set the id of the event
```
As I told you in comments before, each function must have its description.
Provide a description for the set and another one for the get

luvalverde commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+
+private:
+
+    /*! This variable saves the id of the event */
```
As @wfallascr has told you in other comments, it is clear that these are variables, therefore you don't need to specify that in your comments (This variable saves ...)

luvalverde commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+    /*! This variable saves the path of the image of the event */
+    ///@TODO Change the name of the variable to imageEvent when the database is ready
+    QString pathImageEvent;
+
```
remove extra blank lines

luvalverde commented on localMatchApp/headers/ClassEvent.h:
```diff
@@ -0,0 +1,108 @@
+    QString finalDateEvent;
+
+    /*! This variable saves the path of the image of the event */
+    ///@TODO Change the name of the variable to imageEvent when the database is ready
```
Remove this TODO and change the name of the variable.
The TODO should be inside the code where we need to change the path of the image for the image provided by the DB

luvalverde commented on localMatchApp/headers/FunctionsDatabase.h:
```diff
@@ -0,0 +1,67 @@
+
+  /*! Copy Assigment of class FunctionsDatabase */
+
+
```
check all your extra blank lines

luvalverde commented on localMatchApp/headers/FunctionsDatabase.h:
```diff
@@ -0,0 +1,67 @@
+  QString finalDateEvent;
+
+  /*! Declaration of the variable that saves the path of the image of the event */
+  ///@TODO Change the name of the variable to imageEvent when the database is r
```
same comment I gave you before in your other class

luvalverde commented on localMatchApp/headers/homewindow.h:
```diff
@@ -2,24 +2,99 @@
+    ~HomeWindow();
+
+    /*! Copy Constructor of class HomeWindow */
+
```
check all your extra blank lines in your PR

luvalverde reviewed (changes requested):
Please avoid creating PR this huge.
It contains a lot of files and modifications.
The bigger the PR, the harder to catch errors in the code.

And please stop creating a lot of commit messages.
You need to get familiar creating PRs with only the necessarry commits

masolano committed:
Update user interface,class and functions

Update the graphic user interface with the functionality of carousel and add new class and functions of the system with new documentation, constructors and destructors

masolano commented:
PR back to you

luvalverde commented on localMatchApp/headers/GeneralEvent.h:
```diff
@@ -0,0 +1,118 @@
+#include <QWidget>
+
+/*!
+ * The class of General Event is created to
```
Please remove all "The class of ..." from all your classes description.

luvalverde commented on localMatchApp/headers/GeneralEvent.h:
```diff
@@ -0,0 +1,118 @@
+    /*! It saves the path of the image of the event */
+    ///@TODO Change the name of the variable to imageEvent when the database is ready
+    QString imageEvent;
+
```
Be careful with your blank lines. You always have some

luvalverde commented on localMatchApp/headers/homewindow.h:
```diff
@@ -2,24 +2,90 @@
+     * The function when the user click the button of previous event is declare
+     */
+    void onButtonPreviousClicked();
+    /*!
```
add blank lines between functions

luvalverde commented on localMatchApp/headers/homewindow.h:
```diff
@@ -2,24 +2,90 @@
+    void onPushButtonSubscribeClicked();
+
+private:
+    std::unique_ptr<Ui::HomeWindow> ui;
```
missing description

wfallascr commented on localMatchApp/headers/FunctionsDatabase.h:
```diff
@@ -0,0 +1,52 @@
+ * be able to develop the functions
+ * that will manage the database
+ */
+class FunctionsDatabase {
```
This name is redundant. A class is composed of functions and variables. Think of something different, like DataBaseManager, etc

luvalverde commented on localMatchApp/headers/mainwindow.h:
```diff
@@ -3,34 +3,49 @@
 };
 
 #endif // MAINWINDOW_H
+
```
remove blank lines

luvalverde commented on localMatchApp/headers/signup.h:
```diff
@@ -2,31 +2,42 @@
- private:
-  Ui::SignUp *ui;
+private:
+    std::unique_ptr<Ui::SignUp> ui;
```
missing description

wfallascr commented on localMatchApp/headers/FunctionsDatabase.h:
```diff
@@ -0,0 +1,52 @@
+
+};
+
+#endif // FUNCTIONS_H
```
Out of date

wfallascr commented on localMatchApp/headers/GeneralEvent.h:
```diff
@@ -0,0 +1,118 @@
+
+};
+
+#endif // EVENTS_H
```
Out of date

wfallascr commented on localMatchApp/headers/signup.h:
```diff
@@ -2,31 +2,42 @@
+  explicit SignUp (QWidget *i_parent = nullptr);
   ~SignUp ( );
 
+    SignUp (const SignUp& signUp) = default;
```
Indentation problem

luvalverde commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,152 @@
+    /*! If the event is the last, the button next is disabled */
+    /*! To the index of the carousel is substracted a number because the total of events
+    * is a number superior */
+    if(indexOfCarousel==(event.getTotalEvents()-1))
```
Please merge the 3 IF statements into just one to enable/disable the buttons.
You just need to compare if there is just one event, or if we are in the first or last event.
So, with an IF, ELSE IF, ELSE statements you can do the same

wfallascr commented on localMatchApp/sources/homewindow.cpp:
```diff
@@ -1,50 +1,152 @@
 
 /*!
- * In this function is created what is necessary
+ * Is created what is necessary
```
No sentence begin with "is", you are missing the subject

masolano commented:
PR back to you

luvalverde reviewed (approved):
Just change your commit message

masolano commented:
Ready

luvalverde commented:
Please keep the commit header description brief and you then can explain yourself in the commit body message @masolano

masolano commented:
Ready

luvalverde reviewed (approved):
Let's wait for @wfallascr review/approval
